features/cellular/framework/AT/ATHandler.cpp

```diff
@@ -926,16 +926,14 @@ bool ATHandler::consume_to_tag(const char *tag, bool consume_tag)
         int c = get_char();
         if (c == -1) {
             break;
-        } else if (c == tag[match_pos]) {
+        } else if (c == tag[match_pos] || (match_pos = 1 && c == tag[--match_pos])) {
```

Comment:
@AnttiKauppila there is compile warning about the "match_pos = 1": operation on 'match_pos' may be undefined.
Is that something we have to avoid? 
The alternative way which I tried to avoid is an else branch that in case c == tag[0] would have to do same operations as in this branch.

Reply:
This is still causing a warning when compiling with IAR. Could be that we are using version 8 of IAR. Anyway, you could also do the assignment directly which does solve the warning for IAR: 
`} else if (c == tag[match_pos] || c == tag[match_pos = 0]) {`